### PR TITLE
Add segmentation UI in Campaigns wizard

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Segmentation UI for Campaigns plugin.

### How to test the changes in this Pull Request:

_Follow testing steps in https://github.com/Automattic/newspack-popups/pull/289_ 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/wizards/popups/views/segmentation/SingleSegment.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+				if ( foundSegment ) {
+					setName( foundSegment.name );
+					setMinPosts( foundSegment.configuration.min_posts );
+					setMaxPosts( foundSegment.configuration.max_posts );
+				}
+			} );
+		}
+	}, [ isNew ] );
+
+	const saveSegment = () => {
```
Comment (dkoo) [inline assets/wizards/popups/views/segmentation/SingleSegment.js:46]: @adekbadek do you think we need some light validation here to ensure that `max_posts` is greater than `min_posts` if both are enabled?

File: assets/wizards/popups/views/segmentation/SingleSegment.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import { useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { find } from 'lodash';
+
+/**
+ * Internal dependencies.
+ */
+import { Button, TextControl, CheckboxControl, Router } from '../../../../components/src';
+
+const { NavLink, useHistory } = Router;
```
Comment (dkoo) [inline assets/wizards/popups/views/segmentation/SingleSegment.js:13]: After running `npm ci && npm run build`, I keep getting a `TypeError: useHistory is not a function` error that prevents me from viewing the Campaigns wizard screens.

**Review by dkoo — COMMENTED** [review]
@adekbadek I wasn't able to get very far with testing due to the TypeError I mentioned below. I think something might be up with `package-lock.json`? I also tried deleting this and `node_modules` and running `npm install`, but after that I get an error with the `format` import from `@wordpress/date` in `SegmentsList.js`.

File: assets/wizards/popups/views/segmentation/SingleSegment.js (hunk 2 above)
Comment (adekbadek) [reply assets/wizards/popups/views/segmentation/SingleSegment.js:13]: Can you also try resetting the npm packages in `assets/components`?

Comment (dkoo) [reply assets/wizards/popups/views/segmentation/SingleSegment.js:13]: That fixes the `TypeError`, but I still get the `format` error:

```
Uncaught (in promise) TypeError: Cannot read property 'format' of undefined

SegmentsList.js:72
```

Comment (adekbadek) [reply assets/wizards/popups/views/segmentation/SingleSegment.js:13]: d52a2cc0388f0ecc0a418b6c4b054a67701e1003 should fix it, let me know!

File: assets/wizards/popups/views/segmentation/SingleSegment.js (hunk 1 above)
Comment (dkoo) [reply assets/wizards/popups/views/segmentation/SingleSegment.js:46]: I see the "save" button is disabled in this case—thanks!

Comment (dkoo) [reply assets/wizards/popups/views/segmentation/SingleSegment.js:46]: Whoops, unresolving because I was wrong (was just missing a title)

**Review by dkoo — COMMENTED** [review]
This is working for me now. I have one suggestion which is more of a UX question than an issue, but currently it's possible to create segments which will never display a campaign—thinking it might be a quick win to avoid that situation with some front-end validation. Otherwise, looks great to me.

**Comment by adekbadek** [comment]
> currently it's possible to create segments which will never display a campaign—thinking it might be a quick win to avoid that situation with some front-end validation

Good idea, but I think this can be tackled separately, as AFAIK we don't have a consistent strategy for showing form validation errors in the plugin. 
@thomasguillot is there one? How should validation errors be displayed in this UI?

File: assets/wizards/popups/views/segmentation/SingleSegment.js (hunk 1 above)
Comment (adekbadek) [reply assets/wizards/popups/views/segmentation/SingleSegment.js:46]: (responded in [review comment](https://github.com/Automattic/newspack-plugin/pull/689#issuecomment-720647687))

**Review by dkoo — APPROVED** [review]

**Pull request merged**